### DISCUSSION TITLE
often got error: prepared statement "sqlx_s_1" already exists

### DISCUSSION THREAD

**Discussion by ZiJiaW** [SeaQL/sea-orm#1656] [General]
## Description

Recently I tried using sea-orm with rocket and made my main function like this:

``` rust
pub async fn set_up_db() -> Result<DatabaseConnection, DbErr> {
    let pgname = ;
    let pguser = ;
    let pgpassword = ;
    let pghost = ;
    let pgport = ;

    let db_url = format!(
        "postgres://{}:{}@{}:{}/{}",
        pguser, pgpassword, pghost, pgport, pgname
    );
    let mut opt = ConnectOptions::new(db_url);
    let db = Database::connect(opt).await?;
    Ok(db)
}

async fn main() -> Result<(), rocket::Error> {
    let db = set_up_db().await.unwrap();
    rocket::build()
        .manage(db)
        .mount("/edu", routes![index])
        .launch()
        .await?;
    Ok(())
}
```
Now I can write some CRUD code in the `index` handler, and it actually works.
But when I stop the server using ctrl+c after some test requests and rerun the project with `cargo run`,  first several requests will get db error:

```
Database(PgDatabaseError { severity: Error, code: "42P05", message: "prepared statement \"sqlx_s_1\" already exists", detail: None, 
hint: None, position: None, where: None, schema: None, table: None, column: None, data_type: None, constraint: None, file: 
Some("prepare.c"), line: Some(475), routine: Some("StorePreparedStatement") })
```

I can even encounter this when I `ctrl+c` and call sea-orm-cli to generate entity files, which is annoying. 

Is there any way to get rid of this?

By the way, wait for several minutes, this problem will disapper. It seems that sea-orm is reusing a session and generates same names for prepared statements.

## Versions

sea-orm = 0.11.2

**Comment by billy1624** [comment]
Hey @ZiJiaW, after some researching. I found it's a problem with the `pgbouncer`.

- https://github.com/launchbadge/sqlx/issues/67#issuecomment-1090027280
- https://github.com/launchbadge/sqlx/pull/1740#issuecomment-1088045794

I'd suggest you to execute CRUD inside transaction and see if that helped.
